File: api/AirSoft.Service/Contracts/Auth/SignIn/SignInResponse.cs
using Store.Service.Contracts.Jwt.Model;
using Store.Service.Contracts.Models;

namespace Store.Service.Contracts.Auth.SignIn;

public class SignInResponse
{
    public SignInResponse(JwtResponse tokenData, UserData user, AuthProfileData profile)
    {
        TokenData = tokenData;
        User = user;
        Profile = profile;
    }

    public JwtResponse TokenData { get; }

    public UserData User { get; }

    public AuthProfileData Profile { get; }
}

File: strikeball-spa/src/app/shared/services/dto-models/auth/sign-in/sign-in-response.ts
import { IProfileAuthData } from '../profile-auth-data';
import { ITokenData } from '../token-data';
import { IUserData } from '../user-data';

export interface ISignInResponse {
  tokenData: ITokenData | null;
  user: IUserData | null;
  profile: IProfileAuthData | null;
}
